src/protego.py

```diff
@@ -0,0 +1,248 @@
+import re
+from datetime import time
+from collections import namedtuple
+from six.moves.urllib.parse import urlparse, urlunparse, ParseResult, quote, unquote
+
+Record = namedtuple('Record', ['field', 'value'])
+RequestRate = namedtuple(
+    'RequestRate', ['requests', 'seconds', 'start_time', 'end_time'])
+
+DISALLOW_DIRECTIVE = ['disallow', 'dissallow', 'dissalow', 'disalow', 'diasllow', 'disallaw']
+ALLOW_DIRECTIVE = ['allow']
+USER_AGENT_DIRECTIVE = ['user-agent', 'useragent', 'user agent']
+SITEMAP_DIRECTIVE = ['sitemap', 'sitemaps', 'site-map']
+CRAWL_DELAY_DIRECTIVE = ['crawl-delay', 'crawl delay']
+REQUEST_RATE_DIRECTIVE = ['request-rate', 'request rate']
+
+
+def is_valid_directive(d):
```

Comment:
Seeing how these constants are used (checking if a string is in them), I’m quite sure a `set` is a better choice than a `list`.

:lipstick: Also, in the implementation of this function, the code may be slightly more readable [using `any`](https://stackoverflow.com/a/30571351/939364), although I’m not 100% sure.

Reply:
I have removed `is_valid_directive` functions, it wasn't necessary.